tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -107,8 +107,34 @@ std::vector<std::vector<Long64_t>> GetFriendEntries(const std::vector<std::pair<
 
    return friendEntries;
 }
+
+////////////////////////////////////////////////////////////////////////
+/// Return the full path of the tree
+static std::string GetTreeFullPath(const TTree &tree)
+{
+   // Case 1: this is a TChain: we get the name out of the first TChainElement
+   if (0 == strcmp("TChain", tree.ClassName())) {
+      auto &chain = dynamic_cast<const TChain&>(tree);
+      auto files = chain.GetListOfFiles();
+      if (files && 0 != files->GetEntries()) {
+         return files->At(0)->GetName();
```

Comment:
The path within the TFile (and the alternative name if any) is in GetTitle()

Reply:
Do you mean of the chain element?
I get a full path name within the file with the "GetName" method: I am a bit confused.

Reply:
Indeed, I mis-remembered and inverted the semantic and the name and title.

Reply:
cool! Thanks a lot for reading the code and suggesting improvements!